### ISSUE TITLE
inconsistent behavior of Cantus ID links

### ISSUE THREAD

**Issue by jacobdgm** [DDMAL/CantusDB#370]
There are several places on the website where a Cantus ID is displayed as a hyperlink. In different but similar situations, however, these links behave differently:

On Source Detail pages for sources in the Bower database (e.g. http://206.12.93.196/source/123931), the page contains a list of sequences contained in the source, formatted as a table. The final column in the table lists the Cantus ID of the sequence. Clicking on this ID takes you to the Chants-by-Cantus-ID page on CantusDB.

In contrast, if you navigate to a source that's not in the Bower database and click "View all chants" in the sidebar, it takes you to the Browse Chants page (e.g. http://206.12.93.196/chants/?source=123610), which contains a similar-looking table to the one on the Source Detail page. There's a column in that table that lists the Cantus ID of each row's chant. Clicking on this ID takes you to the corresponding Cantus ID page on CantusIndex.

This behavior is also observed in OldCantus. Does anyone know why we would want different behaviors for Cantus IDs, depending on whether the ID is associated with a chant vs. with a sequence? CantusIndex appears to have pages for the CantusIDs of sequences, so that can't be why we don't link to CantusIndex in those situations. And if we have a Chants-by-Cantus-ID page on CantusDB, why should we not make use of it for viewing chants?

I'm not sure what to suggest in order to make this more consistent. One option would be to link in both situations to the Chants-by-Cantus-ID page (and perhaps include a link at the top of that page taking you to CantusIndex). The other is to link in both situations to CantusIndex. (In both situations, we keep the Chants-by-Cantus-ID page around, because it is also linked to on the Feast Detail page)

UPDATE: #372 adds a link to the top of each Chants-by-Cantus-ID page that points toward the relevant page on CantusIndex. My **preferred solution** now (at least my aesthetically preferred solution) is to **link to directly to CantusIndex on chant create/edit/proofread pages and at the top of any list of concordances (which include the same information as the CbCID page anyways), and to the Chants-by-Cantus-ID page everywhere else.**

**Comment by jacobdgm** [comment]
slightly related inconsistency:

Different behavior is observed on Chant Detail vs. Sequence Detail pages. On Sequence Detail pages (e.g. http://206.12.93.196/sequence/628334), the Cantus ID is a hyperlink taking you to CantusIndex, while on Chant Detail pages (e.g. http://206.12.93.196/chant/243157), the CantusID is simply plaintext, with no hyperlink. No reason not to link to CantusIndex in both situations, right?

With #371, this is implemented.

**Comment by jacobdgm** [comment]
another slightly related inelegance:

On Sequence Detail pages (see http://206.12.93.196/sequence/628334), there is a list of concordances. This table has a CantusID column. All its values are the same, because the table lists sequences with the same CantusID as the current sequence. This column and these links are also observed in OldCantus (see https://cantus.uwaterloo.ca/node/628334). Is there a good reason not to delete this column, thus making the table less cluttered?

UPDATE: I made an executive decision and removed this column in #374 .

**Comment by annamorphism** [comment]
my 2 cents:
-yes to the added hyperlink on Chant Detail pages! I have at times been annoyed by the fact that it isn't there in the first place, so this is an improvement in my book. 
-I have never used the Chants-by-Cantus-ID page, presumably because it only worked on sequences. But the information is basically the same as that under "Concordances", no? Which is both easier to get to and more complete (because it includes things from other databases.) So I think it would actually be better to make the sequences link to the Index. (it's possible they were just omitted when the Index split off in the first place.)
-Agree that the Cantus ID in the concordance table is redundant, since they are (by definition) all the same.  I also wonder if the "notes" can be shaved off, since their meaning is mysterious anyway...

**Comment by jacobdgm** [comment]
> But the information is basically the same as that under "Concordances", no?

i.e. the "concordances" that are displayed on the Chant/Sequence detail pages? Yes, I believe all the information from Chants-by-Cantus-ID is already displayed in what's displayed on the detail page.
> (because it includes things from other databases.)

what other databases? Unless you mean Cantus Database and Bower, I don't understand.

**Comment by annamorphism** [comment]
maybe I accidentally discovered an OldCantus/NewCantus behaviour difference!

In OldCantus, the chant-detail concordances shows not only results CantusDB but also other related databases like Cantus Planus in Polonia, Spanish Early Music Manuscripts, etc. (Probably this information is itself taken from Cantus Index?).  So we get 127 concordances for chant 243159: https://cantus.uwaterloo.ca/chant/243159
![image](https://user-images.githubusercontent.com/67451875/193351345-d4df6064-57ef-422e-a7d3-bb0743ecf27c.png)

At NewCantus (http://206.12.93.196/chant/243159) we only get 93, all from within the Cantus DB itself. 
![image](https://user-images.githubusercontent.com/67451875/193351593-098a670d-56d7-4651-970a-db72ccc9d329.png)

**Comment by jacobdgm** [comment]
Oh, excellent catch! And by following the "98 Chant(s)" link next to Cantus Database at the top, it brings us to the chants-by-cantus-id view, a view whose existence makes a lot more sense now.

I'll do some looking into getting links set up to those other databases.

**Comment by jacobdgm** [comment]
I think we're good to go here. As far as I know, the links initially mentioned in this issue behave as they do in OldCantus - it's perhaps inelegant, but we probably don't need to make any changes unless someone complains.

Among the other discrepancies mentioned, we've implemented the missing link on the Chant Detail page, and we now pull our concordances from other databases as OldCantus does.

So yes, I think there's nothing that remains to do here. Closing.
